Firestore/Source/Local/FSTLevelDBMigrations.mm

```diff
@@ -87,6 +87,49 @@ static void AddTargetCount(LevelDbTransaction *transaction) {
   transaction->Put([FSTLevelDBTargetGlobalKey key], targetGlobal);
 }
 
+/**
+ * A class representing all migrations to prepare the LevelDB database for use.
+ *
+ * TODO(wilhuff): All the migrations should really be a part of this but to minimize the scope of
+ * this change for now it's constrained to just the new one.
+ */
+class LevelDbSchema {
+ public:
+  explicit LevelDbSchema(leveldb::DB *db) : db_{db} {
+  }
+
+  void DropQueryCache();
+
+ private:
+  void DeleteEverythingWithPrefix(const std::string &prefix);
+
+  leveldb::DB *db_;
+  std::unique_ptr<LevelDbTransaction> transaction_;
+};
+
+void LevelDbSchema::DropQueryCache() {
+  transaction_ = absl::make_unique<LevelDbTransaction>(db_, "Drop Query Cache");
+
+  DeleteEverythingWithPrefix([FSTLevelDBTargetKey keyPrefix]);
+  DeleteEverythingWithPrefix([FSTLevelDBDocumentTargetKey keyPrefix]);
+  DeleteEverythingWithPrefix([FSTLevelDBTargetDocumentKey keyPrefix]);
+
+  transaction_->Commit();
```

Comment:
`LevelDbTransaction` objects can't be reused after `Commit` so I'm stuck having to allocate a new one to make this work. I can't stack allocate them because any of these functions may need to re-allocate.

If I want to pass everything as parameters, if I have to Commit in one of these invocations I have to pass the transaction in-out and also have to pass the DB. It gets ugly. I figured shared state made this cleaner.

An alternative would be to make transactions reusable after commit. I tried that but it gets hairy in the presence of iterators still open against the transaction.

Ultimately I went with just simplifying this to the point that the migrations just always commit. `DeleteEverythingWithPrefix` may commit multiple times.